sanic/server.py

```diff
@@ -842,6 +847,36 @@ def serve(
         loop.close()
 
 
+def bind_socket(host: str, port: int) -> socket:
+    """Create socket and bind to host.
+    :param host: IPv4, IPv6, hostname or unix:/tmp/socket may be specified
+    :param port: IP port number, 0 or None for UNIX sockets
+    :return: socket.socket object
+    """
+    if host.lower().startswith("unix:"):  # UNIX socket
+        name = host[5:]
+        sock = socket.socket(socket.AF_UNIX)
+        if os.path.exists(name) and os.stat(name) == stat.S_ISSOCK:
+            os.unlink(name)
```

Comment:
is this because we can't clean up the socket on server close?

Reply:
Unlinking of existing socket? Standard practice that I didn't pay much attention to.

Reply:
Unlinking of existing socket? Standard practice that I didn't pay much attention to.

Reply:
I don't really have a preference, its inconsistent with what happens when you try to bind to a port that's in use though so I figured I'd mention it

Reply:
Sockets are left over e.g. after kill -9, so removing them is practical, and apparently asyncio's and uvloop's create_unix_server both will overwrite any existing socket this way. Also, neither of them removes the socket on close, so I've added unlinking code for workers=1 case now too. As a result, both modes of Sanic now (1) remove any existing socket before bind and (2) remove the socket after normal shutdown.

Reply:
@abuckenheimer Thanks for your notes/reviews!